### DISCUSSION TITLE
Can't push to own public gist

### DISCUSSION THREAD

**Discussion by vmakieienko** [community/community#22668] [New to GitHub]
<p>Hi, I created and cloned public gist.<br>
Now I am trying to push some new files to it.<br>
The result I am getting is:<br>
! [remote rejected] master -&gt; master (pre-receive hook declined)<br>
error: failed to push some refs to ‘<a href="https://gist.github.com/2ac86f8732dfddb02f6fd90bc8f0412f.git" rel="nofollow noopener">https://gist.github.com/2ac86f8732dfddb02f6fd90bc8f0412f.git</a>’</p>
<p>What do I do wrong?</p>

**Comment by vmakieienko** [comment]
<p>Ok, I played a bit more with it: I can’t add files in sub-dirs. Is this expected?</p>

**Comment by thinkverse** [comment]
<p>Hi there. 👋</p>
<p>What you’re trying to do simply won’t work, since a gist is one file, mostly use to share snippets and such, what you want to be using is a repository, where you can have multiple files in one place, to do that click your avatar in the top right corner, click on <code>Your GitHub profile</code> to take you to the GitHub main site, now click the <code>+</code> icon next to your avatar and choose <code>New repository</code> and go through the steps there. :+1:</p>
<p>You can read more in depth on the documentation, here’s some useful pages.</p>
<ul>
<li><a href="https://docs.github.com/en/github/getting-started-with-github" rel="nofollow noopener">Getting started with GitHub</a></li>
<li><a href="https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github" rel="nofollow noopener">Creating a repository on GitHub</a></li>
</ul>
<p>Hope that is helpful to you. :slightly_smiling_face:</p>

**Comment by vmakieienko** [comment]
<p>Sorry, <a class="mention" href="/thinkverse">@thinkverse</a>, but your answer is not useful. Gists allow multiple files and pull/push operations. I had my reasons to try them instead of standard repos.</p>

**Comment by otter-computer** [comment, marked as answer]
<p>Hi <a class="mention" href="/vmakieienko">@vmakieienko</a>, welcome to the GitHub Support Community!</p>
<p>Yes, this is expected. Gists can contain multiple files but only on the same directory level. As Gists are intended for sharing snippets of code—usually embedded on other sites—they were designed to show all the code on the same page at the same time and can’t contain directories at all.</p>
<p>If your project needs a directory structure then we’d recommend using a regular repository instead.</p>

**Reply by pelson** [reply]
Wish the pre-receive hook was more informative of this. Wasn't easy to track down.

**Comment by vmakieienko** [comment]
<p>Hi <a class="mention" href="/thomasshaped">@thomasshaped</a>, is there any chance this limitation gets clearly communicated via documentation? I have seen instructions on how to push/pull to/from gits and some lines like “this is just another git repo”(not sure where though), but nothing regarding levels limitation.</p>

**Comment by thinkverse** [comment]
<p>Here is the definition from the <a href="https://docs.github.com/en/github/getting-started-with-github/github-glossary#gist" rel="nofollow noopener">GitHub glossary</a> on <strong>Gists</strong>.</p>
<blockquote>
<p>A gist is a shareable file that you can edit, clone, and fork on GitHub. You can make a gist public or secret, although secret gists will be available to anyone with the URL.</p>
</blockquote>
<p>There is also this section about <a href="https://docs.github.com/en/github/writing-on-github/editing-and-sharing-content-with-gists" rel="nofollow noopener">Editing and sharing content with gists</a>. :slightly_smiling_face:</p>

**Comment by vmakieienko** [comment]
<p>From link you provided:</p>
<blockquote>
<p>Gists are actually Git repositories, which means that you can fork or clone any gist, even if you aren’t the original author. You can also view a gist’s full commit history, including diffs.</p>
</blockquote>
<p>I would say “single level only repositories” would be more clear.</p>
